docs/core-concepts/core-sdk/cli/extensibility.md

```diff
@@ -0,0 +1,174 @@
+project. For instance, various Razor view tools in [ASP.NET](https://www.asp.net/) MVC applications fall into this 
+category. 
+
+### Consuming per-project tools
+Consuming these tools requires you to add a `tools` node to your `project.json`. Inside the `tools` node, you reference
+the package in which the tool resides. After running `dotnet restore`, the tool and its dependencies are restored. 
+
+For tools that need to load the build output of the project for execution, there is usually another dependency which is 
+listed under the regular dependencies in the project file. This means that tools that load project's code have two 
+components: 
+
+1. The "tools" main invoker
+2. Any number of other tools that contain the logic to work with 
+
+Why two things? Tools that need to load the build output of a project need to have unified dependency graph with the 
+project they are working. By adding the dependency bit, we enable NuGet to resolve these dependencies as an unified 
+graph. The invoker is there because it needs to reason about the location as well as the frameworks of the dependency 
+tool. The invoker can accept all of the redirection arguments (`-c`, `-o`, `-b`) that the user specify and find the 
+dependency tool; it can also implement any policies for cases where multiple dependency tools exist for multiple 
+frameworks (i.e. does it run all of them, just one etc.) In general, logic can be shared between these two tools any way 
+that is needed. 
+
+Let's review an example of adding a simple tools-only tool to a simple project. Given an example command called 
+`dotnet-api-search` that allows you to search through the NuGet packages for the specified 
+API, here is a console application's `project.json` file that uses that tool:
+
+```json
+{
+    "version": "1.0.0",
+    "compilationOptions": {
+        "emitEntryPoint": true
+    },
+    "dependencies": {
+        "Microsoft.NETCore.App": {
+            "type": "platform",
+            "version": "1.0.0-rc2-*"
+        }
+    },
+    "tools": {
+        "dotnet-api-search": {
+            "version": "1.0.0",
+            "imports": ["dnxcore50"]
+        }
+    },
+    "frameworks": {
+        "netcoreapp1.0": {}
+    }
+}
+```
+
+The `tools` node is structured in a similar way as the `dependencies` node. It needs the package ID of the package 
+containing the tool and its version at the very least. In the example above, we can see that there is another statement, 
+the `imports` one. This influences the tool's restore process and specifies that the tool is also compatible, in 
+addition to any targeted frameworks the tools has, with `dnxcore50` target. For more information you can 
+consult the [project.json reference](). 
+
+### Building tools
+As mentioned, tools are just portable console applications. You would build one as you would build any console application. 
```

Comment:
change to present tense. For example:
"You build one as you build any other console application.... "